### PR TITLE
HHH-10839 Restore some lost backwards compatibility

### PR DESCRIPTION
**Pull request by Sanne**
https://hibernate.atlassian.net/browse/HHH-10839

### CODE REVIEW

**Comment by Sanne** [comment]
Sorry, not ready for merging as apparently I need to fix something in the documentation:

    Note: Hibernate JPA 2 Static-Metamodel Generator 5.2.1-SNAPSHOT
    /home/sanne/workspaces/hibernate/hibernate-core-parent/documentation/src/test/java/org/hibernate/userguide/hql/HQLTest.java:722: error: incompatible types: org.hibernate.Query cannot be converted to org.hibernate.query.Query
			.setParameter( "name", "J%", StringType.INSTANCE );

I'll check that tomorrow, it's very late now.

**Comment by Sanne** [comment]
I've fixed the failing test, but I had to add a new method to the new `Query` interface: please check this is fine to add, I'm not sure you meant to expose the Hibernate Type on it: 9d8b7e2

File: hibernate-core/src/main/java/org/hibernate/Query.java [hunk 1]
```diff
@@ -54,7 +55,7 @@
  */
 @Deprecated
 @SuppressWarnings("UnusedDeclaration")
-public interface Query<R> extends org.hibernate.BasicQueryContract, TypedQuery<R> {
+public interface Query<R> extends org.hibernate.BasicQueryContract, TypedQuery<R>, CommonQueryContract {
```
Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/Query.java]: a minor note, org.hibernate.BasicQueryContract could be removed

**Comment by dreab8** [comment]
@Sanne it looks fine to me

**Comment by Sanne** [comment]
@dreab8 : pushed a new version to get rid of the the redundant interface. Thanks!

**Comment by dreab8** [comment]
The same issue you had with setParameter(String name, Object val, Type typecan) can occur to code using other methods  like Query<R> setParameter(int position, Object val, Type type);

**Pull request merged**